File: src/main/java/com/cocktailsguru/app/cocktail/dto/CocktailDetailDto.java
package com.cocktailsguru.app.cocktail.dto;

import lombok.Data;

import java.util.List;

@Data
public class CocktailDetailDto {
    private Long id;
    private String name;
    private int totalVolume;
    private int alcoVolume;
    private int nonAlcoVolume;
    private String instructions;
    private String garnish;
    private String description;
    private String imageName;
    private CocktailGlassDto glass;
    private CocktailMethodDto method;
    private List<Long> ingredientList;
    private List<Long> similarCocktailList;
    private int numOfFavorite;
    private double calculatedRating;
    private int numRating1;
    private int numRating2;
    private int numRating3;
    private int numRating4;
    private int numRating5;
    private double alcoholVolume;
    private int numPictures;
    private int numComments;
    private int numShowed;

    public CocktailDetailDto() {
    }
}

File: src/main/java/com/cocktailsguru/app/cocktail/dto/detail/CocktailDetailDto.kt
package com.cocktailsguru.app.cocktail.dto.detail

import com.cocktailsguru.app.cocktail.domain.Cocktail

data class CocktailDetailDto(
        val id: Long,
        val name: String,
        val totalVolume: Int,
        val alcoVolume: Int,
        val nonAlcoVolume: Int,
        val instructions: String,
        val garnish: String,
        val description: String,
        val imageName: String,
        val glass: CocktailGlassDto,
        val method: CocktailMethodDto,
        val ingredientList: List<Long>,
        val similarCocktailList: List<Long>,
        val numOfFavorite: Int,
        val calculatedRating: Double,
        val numRating1: Int,
        val numRating2: Int,
        val numRating3: Int,
        val numRating4: Int,
        val numRating5: Int,
        val alcoholVolume: Double,
        val numPictures: Int,
        val numComments: Int,
        val numShowed: Int
) {
    constructor(cocktail: Cocktail) : this(
            cocktail.id,
            cocktail.name,
            cocktail.totalVolume,
            cocktail.alcoVolume,
            cocktail.nonAlcoVolume,
            cocktail.instructions,
            cocktail.garnish,
            cocktail.description,
            cocktail.imageName,
            CocktailGlassDto(cocktail.glass),
            CocktailMethodDto(cocktail.method),
            cocktail.ingredientList.map { it.ingredient.id },
            cocktail.similarCocktailList.map { it.id },
            cocktail.numOfFavorite,
            cocktail.calculatedRating,
            cocktail.numRating1,
            cocktail.numRating2,
            cocktail.numRating3,
            cocktail.numRating4,
            cocktail.numRating5,
            cocktail.alcoholVolume,
            cocktail.numPictures,
            cocktail.numComments,
            cocktail.numShowed
    )
}
